### DISCUSSION TITLE
gccgo - GCC with go support included?

### DISCUSSION THREAD

**Discussion by scaszoo** [Homebrew/discussions#1973] [Casks] [answered]
Hello :)

I would like to use gcc with go support on my Mac. Sadly the current gcc version does not seem to support go?
Is there any plan to include that in future releases?

Best regards

Christoph

**Comment by carlocab** [comment, marked as answer]
From the [`gcc` formula](https://github.com/Homebrew/homebrew-core/blob/e8e5c0305e2fc5e5632ec995e1fe23ed3ac37aae/Formula/gcc.rb#L64-L67):
```
    # We avoiding building:
    #  - Ada, which requires a pre-existing GCC Ada compiler to bootstrap
    #  - Go, currently not supported on macOS
    #  - BRIG
```

As for future plans, I wouldn't hold my breath:

https://github.com/golang/go/issues/463
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=46986

**Reply by scaszoo** [reply]
Thanks! sadly you might be right...

**Reply by labspc** [reply]
Hello :)

Go has been released to version 1.22 and already supports Darwin. But sadly if you wanna use gccgo, you should compile src code by yourself, and these process has many bugs for beginner.  **Is there any plan for brew to create a  package for gccgo?**

link: [setting up and using gccgo, by go offical, https://go.dev/doc/install/gccgo](https://go.dev/doc/install/gccgo)

img: 

![截屏2024-04-08 22 44 39](https://github.com/Homebrew/discussions/assets/47222768/426ec192-ccd9-4f42-9bf1-1b004b2eea58)


and, now go support this arguement
`go build -compiler=gccgo` ,but you should install gccgo and set PATH first.

> eneral-purpose environment variables:
> GO111MODULE
> Controls whether the go command runs in module-aware mode or GOPATH mode.
> The gccgo command to run for 'go build -compiler=gccgo'.
> https://pkg.go.dev/cmd/go#hdr-Compile_and_run_Go_program

Best regards

Lambert
2024.04.08

**Comment by drpaneas** [comment]
I could also make use of this, really simplifies things to have it instead of compiling it on my own.
